src/pcapi/scripts/payment/user_credit.py

```diff
@@ -0,0 +1,49 @@
+from datetime import datetime
+
+from dateutil.relativedelta import relativedelta
+
+from pcapi.core.payments import models as payments_models
+import pcapi.core.payments.conf as deposit_conf
+from pcapi.core.users import models as users_models
+from pcapi.repository import repository
+
+
+def has_celebrated_their_birthday_since_activation(user: users_models.User) -> bool:
+    return user.deposit_activation_date is not None and user.deposit_activation_date <= user.latest_birthday
+
+
+def has_been_recredited(user: users_models.User) -> bool:
+    if len(user.deposit.recredits) == 0:
+        return False
+
+    if user.deposit.recredits[0].recreditType != deposit_conf.RECREDIT_TYPE_AGE_MAPPING[user.age]:
+        return False
+
+    return True
+
+
+def recredit_underage_users() -> None:
+    sixteen_years_ago = datetime.now() - relativedelta(years=16)
+    eighteen_years_ago = datetime.now() - relativedelta(years=18)
+
+    users: list[users_models.User] = (
+        users_models.User.query.filter(users_models.User.has_underage_beneficiary_role)
+        .filter(users_models.User.deposit is not None)
```

Comment:
Je n'ai pas l'impression que ça fait ce que tu veux. D'après un debug (et sauf si j'ai loupé quelque chose), ça donnerait quelque chose comme `SELECT * from "user" WHERE true` (suivi des conditions sur la date de naissance).

```suggestion
        .filter(users_models.User.deposit.isnot(None))
```